### PR TITLE
Alerting: Add limit query parameter to Loki-based ASH api, drop default limit from 5000 to 1000, extend visible time range for new ASH UI

### PR DESCRIPTION
**Pull request by alexweav**
**What is this feature?**

Allows clients to configure the record limit when querying loki. You can now supply a `limit` query parameter, with a value between 0 and 5000, to control the size of the returned page. Limits are hard-capped at 5000.

The default limit, when the query parameter is not present, has been dropped from 5000 to 1000.

**Which issue(s) does this PR fix?**:

n/a

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by JacobsonMT — APPROVED** [review]
LGTM :rocket: 

One NIT to remove commented out code.

File: pkg/services/ngalert/state/historian/loki_http_test.go [hunk 1]
```diff
@@ -145,13 +145,74 @@ func TestLokiHTTPClient(t *testing.T) {
 			now := time.Now().UTC().UnixNano()
 			q := `{from="state-history"}`
 
-			_, err := client.rangeQuery(context.Background(), q, now-100, now)
+			_, err := client.rangeQuery(context.Background(), q, now-100, now, 1100)
+
+			require.NoError(t, err)
+			params := req.lastRequest.URL.Query()
+			require.True(t, params.Has("limit"), "query params did not contain 'limit': %#v", params)
+			// fmt.Sprint(defaultPageSize)
```
[the remark this answers is not in the document]
Comment (alexweav) [reply pkg/services/ngalert/state/historian/loki_http_test.go:153]: 🤦 good catch :)

File: public/app/features/alerting/unified/components/rules/state-history/LokiStateHistory.tsx [hunk 2]
```diff
@@ -83,7 +91,7 @@ const LokiStateHistory = ({ ruleUID }: Props) => {
   const emptyStateMessage =
     totalRecordsCount > 0
       ? `No matches were found for the given filters among the ${totalRecordsCount} instances`
-      : 'No state transitions have occurred in the last 60 minutes';
+      : 'No state transitions have occurred in the last week';
```
Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/components/rules/state-history/LokiStateHistory.tsx]: should we update this comment?

File: public/app/features/alerting/unified/api/stateHistoryApi.ts [hunk 3]
```diff
@@ -1,15 +1,13 @@
...
 import { DataFrameJSON } from '@grafana/data';
 
 import { alertingApi } from './alertingApi';
 
 export const stateHistoryApi = alertingApi.injectEndpoints({
   endpoints: (build) => ({
-    getRuleHistory: build.query<DataFrameJSON, { ruleUid: string; from: number; to?: number }>({
-      query: ({ ruleUid, from, to = getUnixTime(new Date()) }) => ({
+    getRuleHistory: build.query<DataFrameJSON, { ruleUid: string; from?: number; to?: number; limit?: number }>({
+      query: ({ ruleUid, from, to, limit = 100 }) => ({
```
Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/api/stateHistoryApi.ts:8]: why 100 and not 250?

**Review by soniaAguilarPeiron — APPROVED** [review]
FE code LGTM! 🚀

**Review by JacobsonMT — APPROVED** [review]
LGTM

**Pull request merged**